tests/unit/s2n_protocol_preferences_test.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "tls/s2n_connection.h"
+
+#include <s2n.h>
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test config append */
+    {
+        struct s2n_config *config;
+        EXPECT_NOT_NULL(config = s2n_config_new());
+        EXPECT_EQUAL(config->application_protocols.size, 0);
+        size_t prev_size = 0;
+
+        /* should grow the blob with the provided value */
+        EXPECT_SUCCESS(s2n_config_append_protocol_preference(config, (const uint8_t *)"protocol1", 9));
+        EXPECT_EQUAL(config->application_protocols.size, 1 /* len prefix */ + 9);
```

Comment:
You probably want to avoid a magic "9". If you allocate "protocol1" and "protocol2" ahead of time, you can use sizeof(). It's probably also worth using a #define to name "len prefix" so you don't have to comment it every time.

Reply:
fair enough